### PR TITLE
Remove HAVE_STDLIB_H build var

### PR DESCRIPTION
**Pull request by phongn**
Replace automake-style `HAVE_INCLUDE_H` compile flags with `#if __has_include` preprocessor directives.

This PR does so for `stdlib.h` (which should always exist as a standard header, and thus all guards are removed).

### CODE REVIEW

**Comment by bneradt** [comment]
[approve ci]

**Review by JosiahWI — APPROVED** [review]

**Comment by bneradt** [comment]
[approve ci]

File: include/tscore/ink_platform.h [hunk 1]
```diff
@@ -60,7 +60,7 @@
 struct ifafilt;
 #include <net/if.h>
 
-#ifdef HAVE_STDLIB_H
+#ifdef __has_include(<stdlib.h>)
```
Comment (JosiahWI) [inline include/tscore/ink_platform.h:63]:
```suggestion
#if __has_include(<stdlib.h>)
```

**Review by JosiahWI — CHANGES_REQUESTED** [review]

**Review by JosiahWI — APPROVED** [review]

File: include/tscore/ink_platform.h [hunk 2]
```diff
@@ -60,7 +60,7 @@
 struct ifafilt;
 #include <net/if.h>
 
-#ifdef HAVE_STDLIB_H
+#if __has_include(<stdlib.h>)
```
Comment (dragon512) [inline include/tscore/ink_platform.h:63]: When would we ever not have stdlib.h ... it is a standard C header
We should remove the check for this

**Review by dragon512 — CHANGES_REQUESTED** [review]

File: include/tscore/ink_platform.h (hunk 2 above)
Comment (phongn) [reply include/tscore/ink_platform.h:63]: I agree.

Comment (maskit) [reply include/tscore/ink_platform.h:63]: Yeah, I agree with @dragon512 on this one . I can't think of any system that doesn't need stdlib.h. But at the same time, this is to remove an unnecessary check in CMakeLists.txt. If there is any doubt I'm ok with this change.

**Pull request closed without merging**